src/main/kotlin/app/revanced/patches/shared/spoofsignature/BaseSpoofSignaturePatch.kt

```diff
@@ -0,0 +1,59 @@
+package app.revanced.patches.shared.spoofsignature
+
+import app.revanced.patcher.PatchClass
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patches.shared.gms.BaseGmsCoreSupportPatch
+import app.revanced.patches.shared.spoofsignature.fingerprints.CertificateFingerprint
+import app.revanced.patches.shared.spoofsignature.fingerprints.CertificateFingerprint.GET_PACKAGE_NAME_METHOD_REFERENCE
+import app.revanced.patches.shared.spoofsignature.fingerprints.SpoofSignatureFingerprint
+import app.revanced.patches.shared.spoofsignature.fingerprints.SpoofSignatureFingerprint.GET_ORIGINAL_PACKAGE_NAME_METHOD_NAME
+import app.revanced.patches.shared.spoofsignature.fingerprints.SpoofSignatureFingerprint.INTEGRATIONS_CLASS_DESCRIPTOR
+import app.revanced.util.getTargetIndexWithReference
+import app.revanced.util.resultOrThrow
+import com.android.tools.smali.dexlib2.iface.instruction.FiveRegisterInstruction
+
+/**
+ * Spoof the package name used for app signature verification in Android 12+.
+ * It has not been checked whether it can be used in other Google apps such as Google Maps or Google News,
+ * So patch is not included in [BaseGmsCoreSupportPatch].
+ */
+abstract class BaseSpoofSignaturePatch(
+    private val packageName: String,
+    dependencies: Set<PatchClass> = emptySet()
+) : BytecodePatch(
+    description = "Spoof the package name used for app signature verification in Android 12+.",
```

Comment:
ngl I don't know what this string is for but this change makes it follow the same style as patch descriptions:

```suggestion
    description = "Spoofs the package name used for app signature verification in Android 12+.",
```

Reply:
This string explain how to patch work. It change the current name of patched APK, to align it to the stock version (only inside the code of app, not inside the app list of your phone). In this way, the alphanumeric code used to authenticate your connections to Google servers, will be correctly generated to prevents playback buffering.

Reply:
Yep, this doesn't apply to README.md
(Users cannot see this)

Just consider this a comment for anyone checking code